driver/tests.py

```diff
@@ -27,16 +27,27 @@ def translate():
            "misc/tests/benchmarks/gripper/prob01.pddl"]
     subprocess.check_call(cmd, cwd=REPO_ROOT_DIR)
 
+# We need to translate the example task when this module is imported to have the
+# SAS+ file ready for the @pytest.parametrize function below. Translating the
+# task in setup_module() does not work here, because the @pytest.parametrize
+# decorator is executed before setup_module() is called. An alternative would be
+# to use a conftest.py file and call translate() in a pytest_sessionstart()
+# function, but that adds another file and leads to dumping the translator
+# output to the terminal.
+translate()
```

Comment:
Do we need to call `translate()` at all? The function `run_driver()` will translate the task anyway, won't it? It calls `fast-downward.py` with `--keep`, which should translate the task and keep the sas file, so I assume parameters contain the PDDL files and we need the sas-file for something later on? In the previous code, this looks different: we first called `translate()` to get the sas file and then called the planner, implying that the planner is called on the sas file.

In general, what are the changes in this file about? It seems unrelated to the issue. I don't necessarily mind but I don't get what problem this is fixing.

Reply:
I made these changes to have one pytest test case per configuration instead of only a single one for all portfolios. With this change, we'll see the log output of the failing configuration directly instead of the log of all previous configurations.

Reply:
So the goal is to change the tests from PDDL input and a single run to one translate run, followed by one planner run on the SAS+ file for every config? I don't see the change of input file in the diff, where does this happen?

Given the options in the comment, I like the `conftest.py` option more. Doing significant work when importing a module seems like a code smell to me and `pytest_sessionstart` sounds like the right place for setup like this. We recently added a `conftest.py` in `misc/tests/conftest.py`, so this would be consistent. I would add a comment at the top of the file explaining what it is, though. The first time I saw it in the pull request, I wanted to suggest to rename it to a better name, so a comment explaining that the filename is magic would be good. You could copy it from `misc/tests/conftest.py`.

Reply:
The portfolio configs were always tested on SAS+ directly. But now each planner run is its own test case.

Good to merge now?

Reply:
Ahh, I see, this was in `run_driver` before. Looks good to me.